content/source/docs/enterprise/sentinel/import/tfstate.html.md

```diff
@@ -10,103 +10,526 @@ description: |-
+    "null_resource.foo",
+  ]
+}
+```
+
+The following policy would evaluate to `true`, if the resource was in the state:
+
+```python
+import "tfstate"
+
+main = rule { tfstate.resources.null_resource.bar[0].depends_on contains "null_resource.foo" }
+```
+
+### Value: `id`
+
+* **Value Type:** String.
+
+The `id` value within the [resource
+namespace](#namespace-resources-data-sources) contains id of the resource.
+
+As an example, given the following data source:
+
+-> Note that we use a _data source_ here because the
+[`null_data_soruce`][ref-tf-null-data-source] data source gives a static ID,
+which makes documenting the example easier. As previously mentioned, data
+sources share the same namespace as resources, but need to be loaded with the
+`data` key. For more information, see the
+[synopsis](#namespace-resources-data-sources) for the namespace itself.
+
+[ref-tf-null-data-source]: /docs/providers/null/d/data_source.html
+
+```hcl
+data "null_data_source" "foo" {
+  # ...
+}
+```
+
+The following policy would evaluate to `true`:
+
+```python
+import "tfstate"
+
+main = rule { tfstate.data.null_data_source.foo[0].id is "static" }
+```
+
+### Value: `tainted`
+
+* **Value Type:** Boolean.
+
+The `tainted` value within the [resource
+namespace](#namespace-resources-data-sources) is `true` if the resource is
+marked as tainted in Terraform state.
+
+As an example, given the following resource:
+
+```hcl
+resource "null_resource" "foo" {
+  triggers = {
+    foo = "bar"
+  }
+}
+```
+
+The following policy would evaluate to `true`, if the resource was marked as
+tainted in the state:
+
+```python
+import "tfstate"
+
+main = rule { tfstate.resources.null_resource.foo[0].tainted }
+```
+
+## Namespace: Outputs
+
+The **output namespace** represents all of the outputs present within a
+[module](#namespace-module). This would be any outputs that would have been
+saved to state either as the result of a previous apply, or from the pre-plan
+refresh if their values were known at plan-time.
+
+Note that this can be used to fetch both the outputs of the root module, and the
+outputs of any module in the state below the root, allowing one to see outputs
+that have not been threaded to the root module
+
+This namespace is indexed by output name.
+
+### Value: `sensitive`
+
+* **Value Type:** Boolean.
+
+The `sensitive` value within the [output namespace](#namespace-outputs) is
+`true` when the output has been marked as sensitive.
+
+As an example, given the following output:
+
+```hcl
+output "foo" {
+  sensitive = true
+  value     = "bar"
+}
+```
+
+The following policy would evaluate to `true`:
+
+```python
+import "tfstate"
+
+main = rule { tfstate.outputs.foo.sensitive }
+```
+
+### Value: `type`
+
+* **Value Type:** String.
+
+The `type` value within the [output namespace](#namespace-outputs) gives the
+output's type. This will be one of either `string`, `list`, or `map`, which are
+currently the only types available for outputs in Terraform.
```

Comment:
I'd scrub out either (since it's usually used if there are two, not three, possibilities), and end the sentence after map, then say "These are currently the only..."